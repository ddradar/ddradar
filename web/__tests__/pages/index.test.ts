import Oruga from '@oruga-ui/oruga-next'
import { bulmaConfig } from '@oruga-ui/theme-bulma'
import { RouterLinkStub } from '@vue/test-utils'
import { describe, expect, test, vi } from 'vitest'

import { mountAsync } from '~/__tests__/test-utils'
import Page from '~/pages/index.vue'

vi.mock('#app')

describe('Page /', () => {
  test('renders expected menu', async () => {
    // Arrange - Act
    const wrapper = await mountAsync(Page, {
      global: {
        plugins: [[Oruga, bulmaConfig]],
        stubs: { NuxtLink: RouterLinkStub },
      },
    })

    // Assert
    expect(wrapper.element).toMatchSnapshot()
  })
})
