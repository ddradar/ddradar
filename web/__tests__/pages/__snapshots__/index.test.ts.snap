// Vitest Snapshot v1

exports[`Page / > renders expected menu 1`] = `
<section
  class="section"
>
  <h2
    class="subtitle"
  >
    曲/譜面を探す
  </h2>
  <div
    class="columns is-multiline"
  >
    
    <section
      class="column is-half-tablet is-one-third-desktop is-one-quarter-widescreen"
    >
      <div
        class="o-clps card"
        data-v-f883d62a=""
      >
        <div
          class="o-clps__trigger"
        >
          <header
            aria-controls="曲名から探す"
            class="card-header is-primary"
            data-v-f883d62a=""
            role="button"
          >
            <div
              class="card-header-title"
              data-v-f883d62a=""
            >
              曲名から探す
            </div>
            <a
              class="card-header-icon"
              data-v-f883d62a=""
            >
              <span
                class="icon"
                data-v-f883d62a=""
              >
                <i
                  class="mdi mdi-menu-down mdi-24px"
                />
              </span>
            </a>
          </header>
        </div>
        <transition-stub>
          <div
            class="o-clps__content"
            id="曲名から探す"
            style="display: none;"
          >
            
            <div
              class="card-content"
            >
              <div
                class="buttons"
              >
                
                <a
                  class="button is-text"
                >
                  あ
                </a>
                <a
                  class="button is-text"
                >
                  か
                </a>
                <a
                  class="button is-text"
                >
                  さ
                </a>
                <a
                  class="button is-text"
                >
                  た
                </a>
                <a
                  class="button is-text"
                >
                  な
                </a>
                <a
                  class="button is-text"
                >
                  は
                </a>
                <a
                  class="button is-text"
                >
                  ま
                </a>
                <a
                  class="button is-text"
                >
                  や
                </a>
                <a
                  class="button is-text"
                >
                  ら
                </a>
                <a
                  class="button is-text"
                >
                  わ
                </a>
                <a
                  class="button is-text"
                >
                  A
                </a>
                <a
                  class="button is-text"
                >
                  B
                </a>
                <a
                  class="button is-text"
                >
                  C
                </a>
                <a
                  class="button is-text"
                >
                  D
                </a>
                <a
                  class="button is-text"
                >
                  E
                </a>
                <a
                  class="button is-text"
                >
                  F
                </a>
                <a
                  class="button is-text"
                >
                  G
                </a>
                <a
                  class="button is-text"
                >
                  H
                </a>
                <a
                  class="button is-text"
                >
                  I
                </a>
                <a
                  class="button is-text"
                >
                  J
                </a>
                <a
                  class="button is-text"
                >
                  K
                </a>
                <a
                  class="button is-text"
                >
                  L
                </a>
                <a
                  class="button is-text"
                >
                  M
                </a>
                <a
                  class="button is-text"
                >
                  N
                </a>
                <a
                  class="button is-text"
                >
                  O
                </a>
                <a
                  class="button is-text"
                >
                  P
                </a>
                <a
                  class="button is-text"
                >
                  Q
                </a>
                <a
                  class="button is-text"
                >
                  R
                </a>
                <a
                  class="button is-text"
                >
                  S
                </a>
                <a
                  class="button is-text"
                >
                  T
                </a>
                <a
                  class="button is-text"
                >
                  U
                </a>
                <a
                  class="button is-text"
                >
                  V
                </a>
                <a
                  class="button is-text"
                >
                  W
                </a>
                <a
                  class="button is-text"
                >
                  X
                </a>
                <a
                  class="button is-text"
                >
                  Y
                </a>
                <a
                  class="button is-text"
                >
                  Z
                </a>
                <a
                  class="button is-text"
                >
                  数字・記号
                </a>
                
              </div>
            </div>
            
          </div>
        </transition-stub>
      </div>
    </section>
    <section
      class="column is-half-tablet is-one-third-desktop is-one-quarter-widescreen"
    >
      <div
        class="o-clps card"
        data-v-f883d62a=""
      >
        <div
          class="o-clps__trigger"
        >
          <header
            aria-controls="シリーズから探す"
            class="card-header is-primary"
            data-v-f883d62a=""
            role="button"
          >
            <div
              class="card-header-title"
              data-v-f883d62a=""
            >
              シリーズから探す
            </div>
            <a
              class="card-header-icon"
              data-v-f883d62a=""
            >
              <span
                class="icon"
                data-v-f883d62a=""
              >
                <i
                  class="mdi mdi-menu-down mdi-24px"
                />
              </span>
            </a>
          </header>
        </div>
        <transition-stub>
          <div
            class="o-clps__content"
            id="シリーズから探す"
            style="display: none;"
          >
            
            <div
              class="card-content"
            >
              <div
                class="buttons"
              >
                
                <a
                  class="button is-text"
                >
                  1st
                </a>
                <a
                  class="button is-text"
                >
                  2ndMIX
                </a>
                <a
                  class="button is-text"
                >
                  3rdMIX
                </a>
                <a
                  class="button is-text"
                >
                  4thMIX
                </a>
                <a
                  class="button is-text"
                >
                  5thMIX
                </a>
                <a
                  class="button is-text"
                >
                  DDRMAX
                </a>
                <a
                  class="button is-text"
                >
                  DDRMAX2
                </a>
                <a
                  class="button is-text"
                >
                  EXTREME
                </a>
                <a
                  class="button is-text"
                >
                  SuperNOVA
                </a>
                <a
                  class="button is-text"
                >
                  SuperNOVA2
                </a>
                <a
                  class="button is-text"
                >
                  X
                </a>
                <a
                  class="button is-text"
                >
                  X2
                </a>
                <a
                  class="button is-text"
                >
                  X3 VS 2ndMIX
                </a>
                <a
                  class="button is-text"
                >
                  2013
                </a>
                <a
                  class="button is-text"
                >
                  2014
                </a>
                <a
                  class="button is-text"
                >
                  A
                </a>
                <a
                  class="button is-text"
                >
                  A20
                </a>
                <a
                  class="button is-text"
                >
                  A20 PLUS
                </a>
                <a
                  class="button is-text"
                >
                  A3
                </a>
                
              </div>
            </div>
            
          </div>
        </transition-stub>
      </div>
    </section>
    <section
      class="column is-half-tablet is-one-third-desktop is-one-quarter-widescreen"
    >
      <div
        class="o-clps card"
        data-v-f883d62a=""
      >
        <div
          class="o-clps__trigger"
        >
          <header
            aria-controls="レベルから探す(SINGLE)"
            class="card-header is-primary"
            data-v-f883d62a=""
            role="button"
          >
            <div
              class="card-header-title"
              data-v-f883d62a=""
            >
              レベルから探す(SINGLE)
            </div>
            <a
              class="card-header-icon"
              data-v-f883d62a=""
            >
              <span
                class="icon"
                data-v-f883d62a=""
              >
                <i
                  class="mdi mdi-menu-down mdi-24px"
                />
              </span>
            </a>
          </header>
        </div>
        <transition-stub>
          <div
            class="o-clps__content"
            id="レベルから探す(SINGLE)"
            style="display: none;"
          >
            
            <div
              class="card-content"
            >
              <div
                class="buttons"
              >
                
                <a
                  class="button is-text"
                >
                  1
                </a>
                <a
                  class="button is-text"
                >
                  2
                </a>
                <a
                  class="button is-text"
                >
                  3
                </a>
                <a
                  class="button is-text"
                >
                  4
                </a>
                <a
                  class="button is-text"
                >
                  5
                </a>
                <a
                  class="button is-text"
                >
                  6
                </a>
                <a
                  class="button is-text"
                >
                  7
                </a>
                <a
                  class="button is-text"
                >
                  8
                </a>
                <a
                  class="button is-text"
                >
                  9
                </a>
                <a
                  class="button is-text"
                >
                  10
                </a>
                <a
                  class="button is-text"
                >
                  11
                </a>
                <a
                  class="button is-text"
                >
                  12
                </a>
                <a
                  class="button is-text"
                >
                  13
                </a>
                <a
                  class="button is-text"
                >
                  14
                </a>
                <a
                  class="button is-text"
                >
                  15
                </a>
                <a
                  class="button is-text"
                >
                  16
                </a>
                <a
                  class="button is-text"
                >
                  17
                </a>
                <a
                  class="button is-text"
                >
                  18
                </a>
                <a
                  class="button is-text"
                >
                  19
                </a>
                
              </div>
            </div>
            
          </div>
        </transition-stub>
      </div>
    </section>
    <section
      class="column is-half-tablet is-one-third-desktop is-one-quarter-widescreen"
    >
      <div
        class="o-clps card"
        data-v-f883d62a=""
      >
        <div
          class="o-clps__trigger"
        >
          <header
            aria-controls="レベルから探す(DOUBLE)"
            class="card-header is-primary"
            data-v-f883d62a=""
            role="button"
          >
            <div
              class="card-header-title"
              data-v-f883d62a=""
            >
              レベルから探す(DOUBLE)
            </div>
            <a
              class="card-header-icon"
              data-v-f883d62a=""
            >
              <span
                class="icon"
                data-v-f883d62a=""
              >
                <i
                  class="mdi mdi-menu-down mdi-24px"
                />
              </span>
            </a>
          </header>
        </div>
        <transition-stub>
          <div
            class="o-clps__content"
            id="レベルから探す(DOUBLE)"
            style="display: none;"
          >
            
            <div
              class="card-content"
            >
              <div
                class="buttons"
              >
                
                <a
                  class="button is-text"
                >
                  1
                </a>
                <a
                  class="button is-text"
                >
                  2
                </a>
                <a
                  class="button is-text"
                >
                  3
                </a>
                <a
                  class="button is-text"
                >
                  4
                </a>
                <a
                  class="button is-text"
                >
                  5
                </a>
                <a
                  class="button is-text"
                >
                  6
                </a>
                <a
                  class="button is-text"
                >
                  7
                </a>
                <a
                  class="button is-text"
                >
                  8
                </a>
                <a
                  class="button is-text"
                >
                  9
                </a>
                <a
                  class="button is-text"
                >
                  10
                </a>
                <a
                  class="button is-text"
                >
                  11
                </a>
                <a
                  class="button is-text"
                >
                  12
                </a>
                <a
                  class="button is-text"
                >
                  13
                </a>
                <a
                  class="button is-text"
                >
                  14
                </a>
                <a
                  class="button is-text"
                >
                  15
                </a>
                <a
                  class="button is-text"
                >
                  16
                </a>
                <a
                  class="button is-text"
                >
                  17
                </a>
                <a
                  class="button is-text"
                >
                  18
                </a>
                <a
                  class="button is-text"
                >
                  19
                </a>
                
              </div>
            </div>
            
          </div>
        </transition-stub>
      </div>
    </section>
    <section
      class="column is-half-tablet is-one-third-desktop is-one-quarter-widescreen"
    >
      <div
        class="o-clps card"
        data-v-f883d62a=""
      >
        <div
          class="o-clps__trigger"
        >
          <header
            aria-controls="コースから探す"
            class="card-header is-primary"
            data-v-f883d62a=""
            role="button"
          >
            <div
              class="card-header-title"
              data-v-f883d62a=""
            >
              コースから探す
            </div>
            <a
              class="card-header-icon"
              data-v-f883d62a=""
            >
              <span
                class="icon"
                data-v-f883d62a=""
              >
                <i
                  class="mdi mdi-menu-down mdi-24px"
                />
              </span>
            </a>
          </header>
        </div>
        <transition-stub>
          <div
            class="o-clps__content"
            id="コースから探す"
            style="display: none;"
          >
            
            <div
              class="card-content"
            >
              <div
                class="buttons"
              >
                
                <a
                  class="button is-text"
                >
                  NONSTOP (A20)
                </a>
                <a
                  class="button is-text"
                >
                  段位認定 (A20)
                </a>
                <a
                  class="button is-text"
                >
                  NONSTOP (A20 PLUS)
                </a>
                <a
                  class="button is-text"
                >
                  段位認定 (A20 PLUS)
                </a>
                <a
                  class="button is-text"
                >
                  NONSTOP (A3)
                </a>
                <a
                  class="button is-text"
                >
                  段位認定 (A3)
                </a>
                
              </div>
            </div>
            
          </div>
        </transition-stub>
      </div>
    </section>
    
  </div>
</section>
`;
